### PR TITLE
Show warning when opening an *.old.kdbx file

### LINKED ISSUE #5816: Warn when opening “.old” databases
When “Back up database file before saving,” is enabled, the latest and penultimate versions of a user’s KeePassXC databases are saved right next to each other. This makes it easy to open the “.old” version of the file by accident and to work in it, potentially leading to data loss.

## Expected Behavior

When opening “.old.kdbx” file, KeePassXC should warn the user that they are opening a backup database and not a primary database, which is probably what they intended to do.

## Current Behavior

Opening a “.old” file currently triggers no warning. There are no indications, apart from the double extension of the file being opened that a backup is being opened as opposed to a “live” database.

## Possible Solution

A banner like the one being used to report a failed unlock could be displayed above the unlock form when a backup database is being opened. While the colour of the banner should probably be yellow instead of red to imply a warning as opposed to a failure, the same interface should work well.

## Steps to Reproduce

1. Enable the "Back up database file before saving” setting
2. Work in a database and save it
3. Confirm a “.old” backup has been created
4. Open the backup database in KeePassXC

## Debug info

KeePassXC - Version 2.6.2
Operating system: macOS

### PR DESCRIPTION
**Pull request by termdew**
This PR adds a warning when the user opens an .old.kdbx database file.
Fixes #5816.

## Screenshots
grafik

## Testing strategy
I opened an *.old.kdbx file and looked for a warning message.

## Type of change
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by YoYo-Goal** [comment]
@lightlace ▸ Wow, you’re awesome! I just noticed it seems there is a space missing between the first period and the “Please.” I regret I do not have the knowledge required to prepare a proper Pull Request.

**Comment by lightlace** [comment]
@YoYo-Goal Thanks :) I fixed the spacing in the message, now it should be correct.

File: src/gui/DatabaseOpenWidget.cpp [hunk 1]
```diff
@@ -137,6 +137,14 @@ void DatabaseOpenWidget::load(const QString& filename)
     clearForms();
 
     m_filename = filename;
+    if (m_filename.contains(".old.")) {
+        auto* msgBox = new QMessageBox();
```
Comment (phoerious) [inline src/gui/DatabaseOpenWidget.cpp:141]: This should either be stack-allocated or parented to `this`. I would also check explicitly for `endswith(".old.kdbx")`.

File: src/gui/DatabaseOpenWidget.cpp [hunk 2]
```diff
@@ -137,6 +137,14 @@ void DatabaseOpenWidget::load(const QString& filename)
...
+        msgBox->setWindowTitle(tr("Opening a backup file"));
+        msgBox->setText(tr("You are opening a database backup. Please be aware that it could contain old entries."));
```
Comment (phoerious) [inline src/gui/DatabaseOpenWidget.cpp:143]: I would rephrase it to

> It seems that you are opening a backup database, which may not be up to date.
Do you want to open the File.kdbx instead? Yes - No

It should be paired with a check a) whether there is a file with the same name, but without old and b) whether that file has a newer date.

**Review by phoerious — CHANGES_REQUESTED** [review]

File: src/gui/DatabaseOpenWidget.cpp (hunk 2 above)
Comment (michaelk83) [reply src/gui/DatabaseOpenWidget.cpp:143]: `[Open File] [Cancel]` is confusing. Does Open File open the new file or the old? Does Cancel open the old one, or not open anything? What if the user actually does want to open the old file?
I suggest instead `[Open Old] [Open New] [Cancel]`, where cancel doesn't open anything.

Comment (michaelk83) [reply src/gui/DatabaseOpenWidget.cpp:143]: One more thought: maybe display a warning banner inside the main window while a backup is open. Similar to the "unstable version" warning.
